### PR TITLE
issue 4730 update: check for new versions

### PR DESCRIPTION
**Pull request by hcvcastro**
- infobar: remove unused functions
- infobar: show a snackbar message
- infobar: clean up unused code
- infobar: fire 'infobar' message
- wsd: infobar: add config value "fetch_update_check"
- wsd: infobar: fetch and save the latest version data
- wsd: infobar: send latest version to the client side
- wsd: infobar: clean up unused code

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: wsd/COOLWSD.cpp [hunk 1]
```diff
@@ -4879,6 +4879,51 @@ class COOLWSDServer
...
+        http::Request request(uriFetch.getPathAndQuery());
+
+        if (!sessionFetch || !socketPoll)
+            return;
+
+        request.add("Accept", "application/json");
+        http::Session::FinishedCallback fetchCallback =
+            [sessionFetch, socketPoll](const std::shared_ptr<http::Session>& httpSession)
+        {
+            const std::shared_ptr<const http::Response> httpResponse = httpSession->response();
```
Comment (mmeeks) [inline wsd/COOLWSD.cpp:4898]: This seems to assume that the POCO config is thread-safe - lets have some locking around this - and ideally not use the poco config to store the information.

File: wsd/COOLWSD.cpp [hunk 2]
```diff
@@ -5094,6 +5139,7 @@ int COOLWSD::innerMain()
             = std::chrono::duration_cast<std::chrono::milliseconds>(timeNow - stampFetch);
         if (fetchUpdateCheck > std::chrono::milliseconds::zero() && fetchUpdateCheck >= durationFetch)
         {
+            processFetchUpdate();
```
Comment (mmeeks) [inline wsd/COOLWSD.cpp:5142]: Do we do this on-start ? if fetchUpateCheck is non-zero?

File: wsd/COOLWSD.cpp [hunk 3]
```diff
@@ -5056,6 +5059,7 @@ int COOLWSD::innerMain()
 #endif
 
     const auto startStamp = std::chrono::steady_clock::now();
+    auto stampFetch = startStamp;
```
Comment (mmeeks) [inline wsd/COOLWSD.cpp:5107]: Lets subtract the durationFetch from that and add say 1 minute - so we fetch early after startup the first time.

**Review by mmeeks — CHANGES_REQUESTED** [review]
Looks like we're going in a good direction - some work on thread safety needed first though.

File: wsd/COOLWSD.cpp (hunk 1 above)
Comment (hcvcastro) [reply wsd/COOLWSD.cpp:4898]: Yes, it is thread safe

File: wsd/COOLWSD.cpp (hunk 2 above)
Comment (hcvcastro) [reply wsd/COOLWSD.cpp:5145]: validation in case user disable

File: wsd/COOLWSD.cpp (hunk 3 above)
Comment (hcvcastro) [reply wsd/COOLWSD.cpp]: done

**Review by mmeeks — APPROVED** [review]
Changes are fine - thanks Henry.

Having said that we don't want to break mobile - so we need some #if MOBILE or whatever conditionals around the pieces that cause link errors I think.

Editing the project files there is tough, and of course we really don't need to call-out on mobile to check for updates with the app-store in-place.

Thanks !

**Pull request merged**